ckan/cli/generate.py

```diff
@@ -77,3 +80,44 @@ def extension(output_dir):
 
     cookiecutter(template_loc, no_input=True, extra_context=context,
                  output_dir=output_dir)
+
+
+@generate.command(name=u'config', short_help=u'Create config from template.')
+@click.option(u'-o', u'--output-dir', help=u'Location to put the generated '
+                                           u'template.')
+def make_config(output_dir):
+    if output_dir is None:
+        print('\nERROR: Try again with \'-o path/to/ckan.ini\'')
+        sys.exit(1)
+
+    name = 'ckan.ini'
+
+    if name not in output_dir:
+        print('\nERROR: Config file must be named "ckan.ini"')
+        sys.exit(1)
+
+    # Output to current directory if no path is specified
+    if output_dir == name:
+        output_dir = os.getcwd() + '/' + output_dir
+
+    cur_loc = os.path.dirname(os.path.abspath(__file__))
+    os.chdir(cur_loc)
+    os.chdir('../config')
+    template_loc = os.getcwd() + '/deployment.ini_tmpl'
+    template_variables = {
+        'app_instance_uuid': uuid.uuid4(),
+        'app_instance_secret': ''.join(
+            random.SystemRandom().choice(string.ascii_letters + string.digits)
```

Comment:
The original [paste implementation](https://github.com/cdent/pastescript/blob/e2741855d6250c8f6f942601b0e66e76f3dba564/paste/script/util/secret.py#L24) of this secret encodes it as base64. To be on the safe side in terms of compatibility let's do it as well before passing it to the template.

Reply:
How about using `secrets` library(build-in for py3 and backported as external `python2-secrets`) - it has `token_urlsafe` function that generates base64 encoded token in more secure way than systemRandom does and it's kind of de-facto standard when you are working with all kinds of tokens in py3

Reply:
Sounds fantastic

Reply:
Great! I switched to `token_urlsafe`. 

I didn't think to ask before I pushed the changes, but does `beaker.session.secret` only take alphanumeric characters? In the paste implementation code, it looks like they're only avoiding characters in `'\n\r='`. From the few examples of `beaker.session.secret`s that I have seen, they seem to be alphanumeric though, not including `-` or `_` (which `token_urlsafe` can output).

Reply:
@amercader @smotornyuk I pushed the new changes from the review. Let me know what you think.

Reply:
I think you're happy and I'll just agree that what you have done now is great. Read on for my workings, only in case that's of interest.

It looks like the old method included `[a-z][A-Z][0-9]/+`:
```
$ for i in {1..20}; do paster make-config ckan /tmp/$i.ini; done
$ for i in {1..20}; do grep 'beaker.session.secret' /tmp/$i.ini; done
beaker.session.secret = wwHtgWpAdXp/aGh2jCu09284V
beaker.session.secret = 01kwxI9G1umxh3qJx8ae+ksRz
beaker.session.secret = yMj5cEcVsmLgQhsdp71qEwFyG
beaker.session.secret = gFvxgpVOWvKMWCRAaYDnAfnXt
beaker.session.secret = C5AjGeTEGLnJ2BykE6XmjpcaS
beaker.session.secret = CPsdQg8eXyMGSRVaN2cbB+QUd
beaker.session.secret = 54+5SOpiSghC9fgexjHt4dMNw
beaker.session.secret = 8N/7y/i6j08otGKEpoUIvnCzo
beaker.session.secret = jhnZ9Flpxn9MdiCbO5tD1QppY
beaker.session.secret = WVJwrZcg6JU2AmUCy2xz8jdAL
beaker.session.secret = 8zcVTl/klx1Q8qRESe8Cll1T/
beaker.session.secret = zUXW1EsAbdRbvn7nmakwN+h/5
beaker.session.secret = +BjEzx63gcvSHpQFZMi8tF3mK
beaker.session.secret = CUHPuLDv4tqBb+apYMLQ2wWuQ
beaker.session.secret = eRBPUGNK+5iTG+IlS0z8mqY7u
beaker.session.secret = arYV8bSU3E8puzPCk344UJEaG
beaker.session.secret = mmK40LPvnngWxp1yv2zqQrowV
beaker.session.secret = orUEjHYRY4GqPnUwLi4upDF+9
beaker.session.secret = x5PDimCHWbewwSwsbKURFESSH
beaker.session.secret = 3RXkEjuQslPIi/M3OaO9w/gQ+
```
I really don't think it matters about what symbols are included. As long as its ascii and has no newlines it can be part of the config file. Beaker just wants any ["randomly generated string"](https://beaker.readthedocs.io/en/latest/configuration.html#session-options).

Reply:
Thanks for the information! It's good to know.

Ah, so it looks like `token_urlsafe` will work, though I can easily go with the same or similar implementation paste has. I'll leave it for now, unless anyone thinks there's a reason to have the extra characters.